libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -529,7 +529,7 @@ string IRGenerator::generateFunctionWithModifierInner(FunctionDefinition const&
 		for (size_t i = 0; i < retParams.size(); ++i)
 		{
 			retParamsIn.emplace_back(m_context.newYulVariable());
-			assignRetParams += retParams.back() + " := " + retParamsIn[i] + "\n";
+			assignRetParams += retParams[i] + " := " + retParamsIn[i] + "\n";
```

Comment:
```suggestion
			assignRetParams += retParams[i] + " := " + retParamsIn.at(i) + "\n";
```

or like in line 441, but complementary.

```suggestion
			assignRetParams += retParams[i] + " := " + retParamsIn.back() + "\n";
```

Reply:
What about using `retParams.at(i) + " := " + retParamsIn.at(i) + "\n";` in both places?